dygraphs/dygraphs.d.ts

```diff
@@ -0,0 +1,1272 @@
+     * does not affect the values that appear on tick marks next to the axes. To format those,
+     * see axisLabelFormatter. This is usually set on a <a href='per-axis.html'>per-axis</a>
+     * basis.
+     */
+    valueFormatter?: (
+      v: number,
+      opts: (name: string) => any,
+      seriesName: string,
+      dygraph: Dygraph,
+      row: number,
+      col: number) => any;
+
+    /**
+     * Explicitly set the vertical range of the graph to [low, high]. This may be set on a
+     * per-axis basis to define each y-axis separately. If either limit is unspecified, it will
+     * be calculated automatically (e.g. [null, 30] to automatically calculate just the lower
+     * bound)
+     */
+    valueRange?: number[];
+
+    /**
+     * Whether to display gridlines in the chart. This may be set on a per-axis basis to define
+     * the visibility of each axis' grid separately.
+     */
+    drawGrid?: boolean;
+
+    /**
+     * Show K/M/B for thousands/millions/billions on y-axis.
+     */
+    labelsKMB?: boolean;
+
+    /**
+     * Show k/M/G for kilo/Mega/Giga on y-axis. This is different than <code>labelsKMB</code> in
+     * that it uses base 2, not 10.
+     */
+    labelsKMG2?: boolean;
+  }
+
+  interface Options extends PerSeriesOptions, PerAxisOptions {
+    /**
+     * Set this option to animate the transition between zoom windows. Applies to programmatic
+     * and interactive zooms. Note that if you also set a drawCallback, it will be called several
+     * times on each zoom. If you set a zoomCallback, it will only be called after the animation
+     * is complete.
+     */
+    animatedZooms?: boolean
+
+    /**
+     * If provided, this function is called whenever the user clicks on an annotation.
+     */
+    annotationClickHandler?: (annotation: dygraphs.Annotation, point: Point, dygraph: Dygraph, event: MouseEvent) => any;
+
+    /**
+     * If provided, this function is called whenever the user double-clicks on an annotation.
+     */
+    annotationDblClickHandler?: (annotation: dygraphs.Annotation, point: Point, dygraph: Dygraph, event: MouseEvent) => any;
+
+    /**
+     * If provided, this function is called whenever the user mouses out of an annotation.
+     */
+    annotationMouseOutHandler?: (annotation: dygraphs.Annotation, point: Point, dygraph: Dygraph, event: MouseEvent) => any;
+
+    /**
+     * If provided, this function is called whenever the user mouses over an annotation.
+     */
+    annotationMouseOverHandler?: (annotation: dygraphs.Annotation, point: Point, dygraph: Dygraph, event: MouseEvent) => any;
+
+    /**
+     * Defines per-axis options. Valid keys are 'x', 'y' and 'y2'. Only some options may be set
+     * on a per-axis basis. If an option may be set in this way, it will be noted on this page.
+     * See also documentation on <a href='http://dygraphs.com/per-axis.html'>per-series and
+     * per-axis options</a>.
+     */
+    axes?: {[axis: string]: PerAxisOptions};
```

Comment:
Since there are only 3 valid keys, this could be an object with optional properties `{ x?: PerAxisOptions, y?: PerAxisOptions, y2?: PerAxisOptions }`